onnx/reference/ops/op_constant.py

```diff
@@ -88,27 +88,59 @@ def __init__(self, onnx_node, run_params):  # type: ignore
         if hasattr(self, "sparse_value") and self.sparse_value is not None:  # type: ignore
             self.cst_name = "sparse_value"
             self.cst = self.sparse_value  # type: ignore
+            self.cst_convert = lambda v: v
         elif hasattr(self, "value_float") and self.value_float is not None:  # type: ignore
             self.cst_name = "value_float"
-            self.cst = np.array(self.value_float, dtype=np.float32)  # type: ignore
+            self.cst = (
+                self.value_float  # type: ignore
+                if isinstance(self.value_float, RefAttrName)  # type: ignore
+                else np.array(self.value_float, dtype=np.float32)  # type: ignore
+            )
+            self.cst_convert = lambda v: np.array(v, dtype=np.float32)
```

Comment:
I feel these code is too redundant
How about code like:
```python
for attr, np_dtype in {"value_float": np.float32, "value_floats": np.float32, ...}.items():
    hasattr(self, attr) and getattr(self, attr) is not None:  # type: ignore
            self.cst_name = attr
            v = getattr(self, attr)
            self.cst = (
                v  # type: ignore
                if isinstance(v, RefAttrName)  # type: ignore
                else np.array(v, dtype=np_dtype)  # type: ignore
            )
            self.cst_convert = lambda v: np.array(v, dtype=np_dtype)
```

Reply:
done

Reply:
Thank you! 😄